server-cli/src/main/java/org/apache/brooklyn/cli/Main.java

```diff
@@ -827,6 +827,82 @@ public ToStringHelper string() {
         }
     }
 
+    @Command(name = "clean-orphaned-locations", description = "Removes existing orphaned locations")
+    public static class CleanOrphanedLocationsCommand extends BrooklynCommandCollectingArgs {
+
+        @Option(name = { "--localBrooklynProperties" }, title = "local brooklyn.properties file",
+                description = "local brooklyn.properties file, specific to this launch (appending to and overriding global properties)")
+        public String localBrooklynProperties;
+
+        @Option(name = { "--persistenceDir" }, title = "persistence dir",
+                description = "The directory to read persisted state (or container name if using an object store)")
+        public String persistenceDir;
+
+        @Option(name = { "--persistenceLocation" }, title = "persistence location",
+                description = "The location spec for an object store to read persisted state")
+        public String persistenceLocation;
+
+        @Option(name = { "--destinationDir" }, required = true, title = "destination dir",
+                description = "The directory to copy persistence data to without orphaned locations")
+        public String destinationDir;
+
+        @Option(name = { "--destinationLocation" }, title = "persistence location",
+                description = "The location spec for an object store to copy data to")
+        public String destinationLocation;
+
+        @Option(name = { "--transformations" }, title = "transformations",
+                description = "local transformations file, to be applied to the copy of the data before uploading it")
+        public String transformations;
+
+        @Override
+        public Void call() throws Exception {
+            checkNotNull(destinationDir, "orphanedReferencesTmpDir");
+
+            BrooklynLauncher launcher;
+            failIfArguments();
+
+            try {
+                log.info("Retrieving and copying persisted state to " + destinationDir + " without orphaned locations");
+
+
+                PersistMode persistMode = PersistMode.AUTO;
```

Comment:
I wonder if we should use `PersistMode.REBIND` - i.e. fail if there is no persisted state in the given directory.